### DISCUSSION TITLE
Java setup

### DISCUSSION THREAD

**Discussion by ggrangel** [NvChad/NvChad#2196] [Q&A] [answered]
Hi everyone, 

I'm new to NvChad. What exactly do I need to do setup Java LSP? Do I have to follow all the steps shown in [nvim-jdtls plugin](https://github.com/mfussenegger/nvim-jdtls) or is there something I can skip that already comes built-in with NvChad?

**Comment by siduck** [comment]
@ggrangel nvchad doesnt provide any specific configuration for languages ( only for LUA )

**Comment by antoineco** [comment, marked as answer]
At [example_config/configs](https://github.com/NvChad/example_config/tree/v2.0/configs) you will see a file called `lspconfig.lua`. It contains custom configurations for LSP.

To source this file, while keeping NvChad's built-in LSP configuration (mostly for Lua), make sure your `custom/plugins.lua` file contains:

```lua
  {
    "neovim/nvim-lspconfig",
    config = function()
      require "plugins.configs.lspconfig"
      require "custom.configs.lspconfig"
    end,
  },
```

As the example in the aforementioned `lspconfig.lua` file suggests, you can either

- add `jdtls` to the `servers` table and roll with the default configuration. See the lspconfig docs [here](https://github.com/neovim/nvim-lspconfig/blob/deade69789089c3da15237697156334fb3e943f0/lua/lspconfig/server_configurations/jdtls.lua#L128-L158). You can check the defaults in that same file.

- add a directive with your own configuration instead:
    ```lua
    lspconfig.jdtls.setup {
      on_attach = on_attach,
      capabilities = capabilities,
      settings = {
        -- your personal jdtls preferences
      },
    }
    ```

If `jdtls` is not present in the list if you type `:Mason`, you will indeed need to install it yourself.
